### PR TITLE
Add Spotbugs checks and Spotless formatting for misc files

### PR DESCRIPTION
**Pull request by jjaakola-aiven**
The Spotbugs added for static checking, few issues corrected.

Spotless formatting for misc files to remove unwanted whitespaces.

### CODE REVIEW

File: src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java [hunk 1]
```diff
@@ -406,7 +408,7 @@ private static void addTimeoutConfigGroup(final ConfigDef configDef) {
             ConfigDef.Importance.LOW,
             "HTTP Response timeout (seconds). Default is 30 seconds.",
             TIMEOUT_GROUP,
-            groupCounter++,
+            1,
```
Comment (ivanyu) [inline src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java:411]: This counter looks cumbersome here, but it's uniform with the other groups (where there's more than one configuration). Maybe keep it as is?

**Review by ivanyu — CHANGES_REQUESTED** [review]
One comment and maybe take changes to the tests to a separate PR?

File: src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java (hunk 1 above)
Comment (jjaakola-aiven) [reply src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java:411]: I can add the Spotbugs suppression.

**Review by juha-aiven — COMMENTED** [review]
May add to `README.md` instructions on how to run Spotless formatting?

Should checkstyle be removed?

**Comment by jjaakola-aiven** [comment]
@juha-aiven No removal of Checkstyle needed, I did not configure Spotless for Java formatting.

**Comment by juha-aiven** [comment]
> @juha-aiven No removal of Checkstyle needed, I did not configure Spotless for Java formatting.

Ofc the removal is not needed, but do we really need that when we have Spotless? I wonder of the Spotless-formatted code will always be such that Checkstyle also likes it.

File: README.md [hunk 2]
```diff
@@ -26,12 +26,22 @@ To build a distribution:
...
+Integration tests can be run with
 ```bash
 ./gradlew integrationTest
 ```
 and require [Docker](https://www.docker.com/) installed.
 
+# Formatting
+
+Checkstyle is used for verifying code formatting.
+For misc files Spotless is used.
```
Comment (juha-aiven) [inline README.md:38]: How to read this?

Checkstyle verifies code formatting. Spotless verifies others. But `spotlessApply` formats Java? Why couldn't Spotless also verify Java?

Comment (jjaakola-aiven) [reply README.md:38]: No, like written in. Spotless formats misc files.

Comment (juha-aiven) [reply README.md:38]: OK.... But still I think Checkstyle could be removed and Spotless could both format and verify also code formatting.

**Review by juha-aiven — APPROVED** [review]
LGTM

**Pull request merged**